contributors/design-proposals/storage/volume-topology-scheduling.md

```diff
@@ -107,18 +143,27 @@ weights, but will not be included in the initial implementation.
 
 The advantages of this NodeAffinity field vs the existing method of using zone labels
 on the PV are:
-* We don't need to expose first-class labels for every topology domain.
-* Implementation does not need to be updated every time a new topology domain
+* We don't need to expose first-class labels for every topology key.
+* Implementation does not need to be updated every time a new topology key
   is added to the cluster.
 * NodeSelector is able to express more complex topology with ANDs and ORs.
+* NodeAffinity aligns with how topology is represented with other Kubernetes
+  resources.
 
 Some downsides include:
 * You can have a proliferation of Node labels if you are running many different
   kinds of volume plugins, each with their own topology labeling scheme.
+* The NodeSelector is more expressive than what most storage providers will
+  need. Most storage providers only need a single topology key with
+  one or more domains.  Non-hierarchical domains may present implementation
+  challenges, and it will be difficult to express all the functionality
+  of a NodeSelector in a non-Kubernetes specification like CSI.
 
 
 ### Example PVs with NodeAffinity
```

Comment:
nit: Can you include an example where a fully qualified hierarchy is required to uniquely identify the key, e.g. `zone A, rack B`?

Reply:
done